src/sempy_labs/_notebooks.py

```
@@ -284,3 +298,141 @@ def update_notebook_definition(
+    -------
+    pandas.DataFrame
+        A pandas dataframe showing the notebooks that contain the search string in their definitions.
+        The dataframe includes the workspace name, workspace ID, notebook name, and notebook ID.
+    """
+
+    if not workspace:
+        workspace_id = resolve_workspace_id(workspace)
+        workspace_ids = [workspace_id]
+    elif isinstance(workspace, str):
+        workspace_id = resolve_workspace_id(workspace)
+        workspace_ids = [workspace_id]
```

Ruff (0.12.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)